buildx.sh

```diff
@@ -59,12 +53,28 @@ main() {
         PUSH=false
     fi
 
-    sed -e 's/^FROM alpine:.*$/FROM alpine:edge/' Dockerfile > Dockerfile.edge
+    case "$IMAGE" in
+        fs)
+            BUILD_ARGS="--build-arg ALPINE_PACKAGES= --build-arg COMPOSER_PACKAGES="
+            ;;
+        pdo)
+            BUILD_ARGS="--build-arg ALPINE_PACKAGES=php8-pdo_mysql,php8-pdo_pgsql --build-arg COMPOSER_PACKAGES="
+            ;;
+        gcs)
+            BUILD_ARGS="--build-arg ALPINE_PACKAGES=php8-openssl"
+            ;;
+        *)
+            BUILD_ARGS=""
+            ;;
+    esac
+    IMAGE="privatebin/$IMAGE"
 
-    image_build_arguments | while read -r IMAGE BUILD_ARGS ; do
+    if [ "$EDGE" = false ] ; then
```

Comment:
I would switch the conditions here, because otherwise it ~~somewhat looks like [yoda conditions](https://en.wikipedia.org/wiki/Yoda_conditions)~~ somehow sounds strange, given that you explicitly check whether a special build variant is _not_ used. I.e. it sounds complicated… :D

Also if `$EDGE` is unset for some reason, it should obviously do a non-edge build and not an edge build.

Just a minor code-style thing. Do whatever you think is best.

Reply:
It makes sense, if only for consistency.

Reg. the $@ quoting to avoid re-splitting: In that particular case we actually do need the array to be re-split. If we don't, then the `"$BUILD_ARGS"` would not get split and the docker command would fail or at least ignore those arguments, resulting in images that all contain the same thing.

Honestly, the argument splitting done by shells is one of it's design weak-points and in most other cases shellcheck is right to remind about the quoting, to prevent nasty surprises down the line. I have this awesome tool integrated in my IDE as well and it saved me countless of times from the shells many pit falls.